src/main/java/org/ossgang/commons/utils/GeneralDecimalFormat.java

```diff
@@ -0,0 +1,77 @@
+package org.ossgang.commons.utils;
+
+import static java.lang.Math.pow;
+
+import java.text.DecimalFormat;
+import java.text.FieldPosition;
+import java.text.NumberFormat;
+import java.text.ParsePosition;
+
+/**
+ * A Decimal Format that behaves similar to the format string "%g" ("general"): if the number is representable with
+ * a small number of digits, using fixed-point formatting, use fixed point. If it is too small or too big, fall back to
+ * scientific format.
+ */
+public class GeneralDecimalFormat extends NumberFormat {
+    private final DecimalFormat fixedFormat;
+    private final DecimalFormat exponentialFormat;
+    private final double lowExponentialThreshold;
+    private final double highExponentialThreshold;
+
+    public GeneralDecimalFormat(DecimalFormat fixedFormat, DecimalFormat exponentialFormat,
+            double lowExponentialThreshold, double highExponentialThreshold) {
+        if (lowExponentialThreshold <= 0 || highExponentialThreshold <= 0) {
+            throw new IllegalArgumentException("Thresholds must be > 0!");
+        }
+        this.fixedFormat = fixedFormat;
+        this.exponentialFormat = exponentialFormat;
+        this.lowExponentialThreshold = lowExponentialThreshold;
+        this.highExponentialThreshold = highExponentialThreshold;
+    }
+
+    public GeneralDecimalFormat(String pattern, double lowExponentialThreshold, double highExponentialThreshold) {
+        this(new DecimalFormat(pattern), new DecimalFormat(pattern + "E0"), lowExponentialThreshold,
```

Comment:
What if the passed in pattern contains already an 'E0' string. Would then the "+'E0'" result in an invalid pattern? If so, consider throwing in this case ....

Reply:
Indeed, good point (even if it means I have to extract it into a method :smile: )